salt/modules/btrfs.py

```diff
@@ -673,3 +673,465 @@ def properties(obj, type=None, set=None):
+    # spaces. The issue is that we do not know how many spaces do we
+    # have in the path name, so any classic solution based on split
+    # will fail.
+    #
+    # This list is in order.
+    columns = ('ID', 'gen', 'cgen', 'parent', 'top level', 'otime',
+               'parent_uuid', 'received_uuid', 'uuid', 'path')
+    result = []
+    for line in res['stdout'].splitlines():
+        table = {}
+        for key in columns:
+            value, line = _pop(line, key, key == 'path')
+            if value:
+                table[key.lower()] = value
+        # If line is not empty here, we are not able to parse it
+        if not line:
+            result.append(table)
+
+    return result
+
+
+def subvolume_set_default(subvolid, path):
+    '''
+    Set the subvolume as default
+
+    subvolid
+        ID of the new default subvolume
+
+    path
+        Mount point for the filesystem
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' btrfs.subvolume_set_default 257 /var/volumes/tmp
+
+    '''
+    cmd = ['btrfs', 'subvolume', 'set-default', subvolid, path]
+
+    res = __salt__['cmd.run_all'](cmd)
+    salt.utils.fsutils._verify_run(res)
+    return True
+
+
+def subvolume_show(path):
+    '''
+    Show information of a given subvolume
+
+    path
+        Mount point for the filesystem
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' btrfs.subvolume_show /var/volumes/tmp
+
+    '''
+    cmd = ['btrfs', 'subvolume', 'show', path]
+
+    res = __salt__['cmd.run_all'](cmd)
+    salt.utils.fsutils._verify_run(res)
+
+    result = {}
+    table = {}
+    # The real name is the first line, later there is a table of
+    # values separated with colon.
+    stdout = res['stdout'].splitlines()
+    key = stdout.pop(0)
+    result[key.strip()] = table
+
+    for line in stdout:
+        key, value = line.split(':', 1)
+        table[key.lower().strip()] = value.strip()
+    return result
+
+
+def subvolume_snapshot(source, dest=None, name=None, read_only=False):
+    '''
+    Create a snapshot of a source subvolume
+
+    source
+        Source subvolume from where to create the snapshot
+
+    dest
+        If only dest is given, the subvolume will be named as the
+        basename of the source
+
+    name
+       Name of the snapshot
+
+    read_only
+        Create a read only snapshot
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' btrfs.subvolume_snapshot /var/volumes/tmp dest=/.snapshots
+        salt '*' btrfs.subvolume_snapshot /var/volumes/tmp name=backup
+
+    '''
+    if not dest and not name:
+        raise CommandExecutionError('Provide parameter dest, name, or both')
+
+    cmd = ['btrfs', 'subvolume', 'snapshot']
+    if read_only:
+        cmd.append('-r')
+    if dest and not name:
+        cmd.append(dest)
+    if dest and name:
+        name = os.path.join(dest, name)
+    if name:
+        cmd.append(name)
```

Comment:
No need in all these `if if if`:
```python
cmd.append(os.path.join(dest or '', name or '').strip(os.path.sep))
```

Reply:
I am not sure why to favor a more obfuscated code.

I am personally not so smart to realize that the last `.strip(os.path.sep)` is needed because if `name` is not assigned, `os.path.join(dest, '')` will add this unneeded extra '/'.

Let me be easy and explicit here!